### PR TITLE
Update to GHC 9.6

### PR DESCRIPTION
**Pull request by blishko**
## Description
This finishes the work started in #471.

## Checklist

- [x] tested locally
- [ ] added automated tests
- [ ] updated the docs
- [ ] updated the changelog

### CODE REVIEW

**Comment by msooseth** [comment]
Let me know when this is ready for review! :) I'd be happy to review & merge.

File: hevm.cabal [hunk 1]
```diff
@@ -108,17 +111,9 @@ library
     Paths_hevm
   autogen-modules:
     Paths_hevm
-  if impl(ghc >= 9.4) && !os(darwin)
-    -- darwin is skipped because it produces this error when building
-    -- > ghc: loadArchive: Neither an archive, nor a fat archive: `/nix/store/l3lkdfm7sg1wwc850451cikqds766h15-clang-wrapper-11.1.0/bin/clang++'
+  if os(linux)
     build-depends: system-cxx-std-lib
-  elif !os(darwin)
     extra-libraries: stdc++
```
Comment (elopez) [inline hevm.cabal:116]: I think we may be able to simplify this now and just have `build-depends: system-cxx-std-lib` for all platforms, plus the special darwin flags just in one place.

Comment (blishko) [reply hevm.cabal:116]: @elopez , do you want to try it?
There are already 4 people contributing to this PR, why not make it 5 😄

Comment (blishko) [reply hevm.cabal]: Your commits are included. First, there was some problem in CI, but after restarting a job, everything is green now!

**Comment by blishko** [comment]
@d-xo, I guess we can merge now, WDYT?

**Comment by elopez** [comment]
@blishko there's one last issue remaining, the redistributable files produced by nix contain references to the nix store due to how c++ / c++abi is now split. I'm hoping this fixes it and prevents the issue from going unnoticed again https://github.com/elopez/hevm/pull/4/commits/6575b73ca87ee6165739d88f2adc47562845a60f

**Comment by d-xo** [comment]
@elopez thats not a regression and I think it can be fixed in a separate pr. Let's merge this now. thanks for the work everyone,I think we have commits from almost every active hevm contributor on this branch ❤️

**Pull request merged**